### PR TITLE
Deploy to Netlify

### LINKED ISSUE #491: Build is broken
https://travis-ci.org/dnsimple/dnsimple-support/jobs/625724721

```
All images optimized!
Publishing to S3
[info] Downloading https://github.com/laurilehmijoki/s3_website/releases/download/v3.4.0/s3_website.jar into /home/travis/build/dnsimple/dnsimple-support/vendor/bundle/ruby/2.6.0/gems/s3_website-3.4.0/s3_website-3.4.0.jar
[fail] Could not load the site: Failed to parse ERB in /home/travis/build/dnsimple/dnsimple-support/s3_website.yml:
       (SyntaxError) /home/travis/.rvm/rubies/ruby-2.6.5/lib/ruby/site_ruby/2.6.0/rubygems/core_ext/kernel_warn.rb:15: syntax error, unexpected tLABEL
           module_function define_method(:warn) {|*messages, uplevel: nil|
                                                                    ^
Published
```

### PR DESCRIPTION
**Pull request by sbastn**
Fixes #491. This PR deploys this repository to Netlify automatically when the `master` branch is updated.

### Highlights
1. Removes `s3_website` gem and configuration files
2. Updates Travis config to no longer deploy. It runs the tests, and compile the pages.
3. Updates README.md
4. Adds canonical link to avoid SEO duplication.

### Verification steps
- Verify that the canonical header is correct when testing this branch locally
- Netlify has been already been configured to deploy the latest on `master`. Check a few pages here: dnsimple-support.netlify.com

### Post steps

- [ ] Configure DNS records to point to dnsimple-support.netlify.com

### CODE REVIEW

**Comment by weppos** [comment]
A couple of questions:

1. What about the `publish` task. Is it still used?
2. How is Netlify interacting with Travis? Is it deploying if the specs fail?
3. The `after_success` task was used to run the publishing. What's the purpose now?

**Comment by weppos** [comment]
Also, before switching we need to verify:

1. We can use a custom domain hence preserve support.dnsimple.com
2. Use an auto-generated SSL certificate for the subdomain
3. Use the HTTPS version by default (hence http should redirect to https)

**Comment by sbastn** [comment]
> What about the publish task. Is it still used?

Yes, it is used by Netlify.

> How is Netlify interacting with Travis? Is it deploying if the specs fail?

I had a couple of build breaks in this PR and it seemed that Netlify wasn't deploying at that moment, but I'll give you a more accurate answer once I reproduce this scenario on purpose.

> The after_success task was used to run the publishing. What's the purpose now?

We're now using `rake` default task, which runs the tests and compile the pages.

> How are the redirects configured?

Through the `_redirects` file.

> We can use a custom domain hence preserve support.dnsimple.com

They do support it, but we have to bring the DNS to them: _You can bring a domain name that you already own, or buy a new one. When you buy the domain with us, we automatically configure your DNS settings and provision a wildcard certificate for your domain._

> Use an auto-generated SSL certificate for the subdomain

Also done through DNS

> Use the HTTPS version by default (hence http should redirect to https)

Using https://https.netlify.com/ could be an option

**Comment by sbastn** [comment]
@weppos I was able to configure support-test.dnsimple.com with a LE certificate and automatic redirection from HTTP to HTTPS:
```
> curl -I support-test.dnsimple.com
HTTP/1.1 301 Moved Permanently
Cache-Control: public, max-age=0, must-revalidate
Content-Length: 50
Content-Type: text/plain
Date: Fri, 03 Jan 2020 12:50:19 GMT
Location: https://support-test.dnsimple.com/
Age: 297
Connection: keep-alive
Server: Netlify
X-NF-Request-ID: 771ff61d-3561-4f6b-bbf9-35d4c6e1874e-6235450
```

I am still investigating if we can avoid deploying if Travis fails the build. Unfortunately, I thought this was working already.

**Comment by sbastn** [comment]
@weppos By moving the `test` rake task as part of the `publish` task, we get Netlify to stop the build & deploy if something is wrong. This makes Travis redundant, we could remove it. Netlify can integrate with Slack to push notifications about Netlify success/failure events.

I think we now check all the boxes. With Netlify, we can:
✅ Keep using our DNS
✅ Enforce HTTPS traffic
✅ Automatically configure a TLS/SSL certificate
✅ Get notifications about the status of a build & deploy
✅ Redirect old URLs

Do you have any other requirement that we should add to the list?

**Comment by sbastn** [comment]
I have fixed an issue with URL redirection.

**Comment by sbastn** [comment]
I have fixed an issue with URL redirection. Everything is working at this point.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,10 @@
 language: ruby
 
-jdk: openjdk8
-
 cache:
   - bundler
 
 after_success:
-  - test $TRAVIS_PULL_REQUEST == "false" && test $TRAVIS_BRANCH == "master" && bundle exec rake publish
+  - test $TRAVIS_PULL_REQUEST == "false" && test $TRAVIS_BRANCH == "master" && bundle exec rake
```
Comment (weppos) [inline .travis.yml]: @sbastn can you explain the reason you kept this config?

Comment (sbastn) [reply .travis.yml]: I am going to remove the travis config since everything is done through Netlify at this point.

**Pull request merged**